### DISCUSSION TITLE
Will the downloader still work when Reddit starts charging for using the API?

### DISCUSSION THREAD

**Discussion by instaloadererror** [Serene-Arc/bulk-downloader-for-reddit#870] [Q&A]
Will the downloader still work when Reddit starts charging for using the API?

**Comment by Serene-Arc** [comment]
So the new changes are predominantly rate limits depending on the client ID of the BDFR. This is a customisable field so the worst case scenario is that the user will have to register their own ID rather than use the pre-written one registered now. I'm investigation the effects on our codebase and what it will affect, but the worst that can happen is that the BDFR is slowed down since there's a limit of 100 requests per minute. We'll have to see when the rate-limiting is put into effect.

**Reply by instaloadererror** [reply]
Will people have to start paying Reddit for it to work?

**Reply by Serene-Arc** [reply]
From my understanding, no. There will be a free tier for authenticated users of 100 requests per minute. This is slower than what we have now I think, but shouldn't make the BDFR unreasonably slow in terms of the download side of things. The cloning and archive functions might be affected but we can look into grouping API calls to try and optimise the number of requests where possible.
